storage/local/series.go

```diff
@@ -377,9 +386,27 @@ func (s *memorySeries) newIterator(pinnedChunkDescs []*chunkDesc) SeriesIterator
 // preloadChunksForRange loads chunks for the given range from the persistence.
 // The caller must have locked the fingerprint of the series.
 func (s *memorySeries) preloadChunksForRange(
+	fp model.Fingerprint,
 	from model.Time, through model.Time,
-	fp model.Fingerprint, mss *memorySeriesStorage,
+	lastSampleOnly bool,
+	mss *memorySeriesStorage,
 ) ([]*chunkDesc, SeriesIterator, error) {
+	// If we have to preload for only one sample, and we have a
+	// lastSamplePair in the series, and thas last samplePair is in the
+	// interval, just take it in a singleSampleSeriesIterator. No need to
+	// pin or load anything.
+	if lastSampleOnly {
```

Comment:
In principle yes, but it's just pass-through from preloadChunksForRange in memorySeriesStorage, and reworking this one is a bit more cumbersome.

I'll give it a try anyway, and then you con decide if you like it more or not.